packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -148,52 +148,101 @@ export class LedgerKeyAgent extends KeyAgentBase {
     this.#communicationType = serializableData.communicationType;
   }
 
-  static findKeyAgentByCommunicationTypeAndDevicePath(communicationType: CommunicationType, devicePath?: string) {
-    return this.deviceConnections?.find(
+  private static async findConnectionByCommunicationTypeAndDevicePath(
+    communicationType: CommunicationType,
+    devicePath?: string
+  ): Promise<LedgerConnection | null> {
+    const matchingConnectionData = this.deviceConnections?.find(
       (connection) => connection.communicationType === communicationType && connection.devicePath === devicePath
     );
+    if (!matchingConnectionData) return null;
+
+    try {
+      await this.testConnection(matchingConnectionData.deviceConnection);
+    } catch (error) {
+      if (
+        error instanceof errors.TransportError &&
+        error.message.includes('Cannot communicate with Ledger Cardano App')
```

Comment:
Would be better to have this as a const defined in 1 place
```suggestion
        error.message === CONNECTION_FAILURE_ERROR_MESSAGE
```

Reply:
I created a `CARDANO_APP_CONNECTION_ERROR_MESSAGE` variable and used it when throwing an error and when catching it.